common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -1078,7 +1078,22 @@ public UTF8String[] split(UTF8String pattern, int limit) {
       }
       return result;
     }
-    return split(pattern.toString(), limit);
+    return split(pattern.toString(), limit, regexFlags);
+  }
+
+  public UTF8String[] split(UTF8String pattern, int limit) {
+    return split(pattern, limit, 0); // Pattern without regex flags
+  }
+
+  public UTF8String[] splitCollationAware(UTF8String pattern, int limit, int collationId) {
+    if (CollationFactory.fetchCollation(collationId).supportsBinaryEquality) {
+      return split(pattern, limit);
+    }
+    if (collationId == CollationFactory.UTF8_BINARY_LCASE_COLLATION_ID) {
+      return split(pattern, limit, Pattern.UNICODE_CASE | Pattern.CASE_INSENSITIVE);
+    }
+    throw new UnsupportedOperationException("Unsupported collation " +
+      CollationFactory.fetchCollation(collationId).collationName);
```

Comment:
if you are using `StringTypeBinaryLcase` for this expression, it should never come to this point
that said, we may want to keep this here anyway because this method is part of a public interface

Reply:
+1, this should fail in inputTypeCheck of the related expression. I am not sure if we need this double guard.